src/consumer.coffee

```diff
@@ -1,7 +1,52 @@
+crypto            = require 'crypto'
+URL         = require 'url-parse'
+HMAC_SHA1         = require './hmac-sha1'
+errors            = require './errors'
+extensions        = require './extensions'
+
 class Consumer
-  constructor: () ->
-    console.error('Consumer Not Implemented')
-    false
+  constructor: (consumer_key, consumer_secret) ->
+    if typeof consumer_key is 'undefined' or consumer_key is null
+      throw new errors.ConsumerError 'Must specify consumer_key'
+
+    if typeof consumer_secret is 'undefined' or consumer_secret is null
+      throw new errors.ConsumerError 'Must specify consumer_secret'
+
+    @consumer_key     = consumer_key
+    @consumer_secret  = consumer_secret
+    @signer           = new HMAC_SHA1()
+    @body             = {}
+
+  sign_request: (url, method = 'GET', body = {}) =>
```

Comment:
The reasoning behind this was to make it simpler to sign requests as a TC since this library is already using crypto, parsing and building the request string in the proper order, and signing it properly. If you think that this won't be useful for the library going forward I can reevaluate how we are doing this on our side. I wanted to throw this into here because I thought it would be useful for others that are looking easily sign these requests. Let me know!